handlers/utils.go

```diff
@@ -22,13 +22,65 @@ const defaultPatchType = "merge"
 
 var defaultPromClientSupplier = prometheus.NewClient
 
-func checkNamespaceAccess(ctx context.Context, nsServ business.NamespaceService, namespace string) (*models.Namespace, error) {
-	return nsServ.GetNamespace(ctx, namespace)
+func checkNamespaceAccess(ctx context.Context, nsServ business.NamespaceService, namespace string, cluster string) (*models.Namespace, error) {
+	return nsServ.GetClusterNamespace(ctx, namespace, cluster)
 }
 
-func createMetricsServiceForNamespace(w http.ResponseWriter, r *http.Request, promSupplier promClientSupplier, namespace string) (*business.MetricsService, *models.Namespace) {
-	metrics, infoMap := createMetricsServiceForNamespaces(w, r, promSupplier, []string{namespace})
-	if result, ok := infoMap[namespace]; ok {
+// createMetricsServiceForNamespaceMC is used when the service will query across all clusters for the namespace.
+// It will return an error if the user does not have access to the namespace on all of the clusters.
+func createMetricsServiceForNamespaceMC(w http.ResponseWriter, r *http.Request, promSupplier promClientSupplier, nsName string) (*business.MetricsService, []models.Namespace) {
+	layer, err := getBusiness(r)
+	if err != nil {
+		RespondWithError(w, http.StatusInternalServerError, err.Error())
+		return nil, nil
+	}
+	prom, err := promSupplier()
+	if err != nil {
+		log.Error(err)
+		RespondWithError(w, http.StatusServiceUnavailable, "Prometheus client error: "+err.Error())
+		return nil, nil
+	}
+	var nsInfo []models.Namespace
+
+	nsClusters, err := layer.Namespace.GetNamespaceClusters(r.Context(), nsName)
+	if err != nil || len(nsClusters) == 0 {
+		log.Infof("No clusters found.")
+		//TODO: We cant find the cluster list, we assume it is local
+		nsClusters = make([]models.Namespace, 1)
+		nsClusters[0] = models.Namespace{Cluster: config.Get().KubernetesConfig.ClusterName, Name: nsName}
+	}
```

Comment:
Unless you can explain in detail why we need this behavior I still think this should just `RespondWithError`.

Reply:
Update to return the clusters name list using the user clients map index. 

Using GetNamespaceClusters was not working in this case because we were returning an error, but we want to get all the whole list and then check the access for each particular namespace.